### PR TITLE
Move onNotFound to options, call it from evaluateCurrent

### LINKED ISSUE #11: Fix default route/not found behavior
The difference between `evaluateCurrent` and `setDefault` is confusing and wrong, and is making life in the abstract-state-router difficult (see https://github.com/TehShrike/abstract-state-router/issues/63).

Right now `evaluateCurrent` checks the current route, and if it is empty, navigates to the default.  If the current route is not empty, then it tries to navigate to it, with the special rule that for that one command, the default route is whatever was passed in to `evaluateCurrent` instead of the usual default route.

Instead of this, `evaluateCurrent` should get the current route, and
- if the route is empty, go to the default
- if the route is not empty, navigate to it, and emit a "not found" event if that url is not matched

Along this vein, I think that `setDefault` isn't the right way to handle invalid urls.  There should probably be an `onNotFound` callback instead.

So to sum up: on `evaluateCurrent`, if there is no url, the default url should be visited.  If a url is visited and has no route, an `onNotFound` callback should be called.

This would be a breaking change.

### PR DESCRIPTION
**Pull request by TehShrike**
One new test.

Fixes #11

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
   },
   "dependencies": {
     "array.prototype.find": "^1.0.0",
+    "noop2": "2.0.0",
```
Comment (saibotsivad) [inline package.json]: Unrelated to this pull request, but this has a fixed version while the other modules are using `^`.

File: index.js [hunk 2]
```diff
@@ -3,12 +3,14 @@ var qs = require('querystring')
 var xtend = require('xtend')
 var browserHashLocation = require('./hash-location.js')
 require('array.prototype.find')
+var noop = require('noop2')
```
Comment (ArtskydJ) [inline index.js]: This line offends me on a philosophical level.

File: package.json (hunk 1 above)
Comment (TehShrike) [inline package.json]: Good point, it wouldn't hurt to make the others a bit more narrowly-scoped

**Comment by saibotsivad** [comment]
👍

**Comment by ArtskydJ** [comment]
:+1:

**Comment by ArtskydJ** [comment]
Aside from lack of docs, lgtm...

**Comment by TehShrike** [comment]
Published as 2.0.0

**Pull request merged**